### ISSUE TITLE
Two enums those have same name and same member names assignable each other.

### ISSUE THREAD

**Issue by kazatsuyu** [microsoft/TypeScript#35138]
**TypeScript Version:**  3.8.0-dev.20191115

**Search Terms:**

**Code**

```ts
module A {
    export enum X { X1, X2 }
}

module B {
    export enum X { X1 = 'X1', X2 = 'X2' }
}

declare const ax: A.X;
const bx: B.X = ax; // It should be an error.
```

**Expected behavior:**
`A.X` and `B.X` are not assignable each other.

**Actual behavior:**
They are assignable.

**Playground Link:** 
https://www.typescriptlang.org/play/?ts=3.8.0-dev.20191115&ssl=1&ssc=1&pln=11&pc=1#code/LYewJgrgNgpgBAQTgbwFBw3GAPADiAJwBcsA7CYOADRWoEYAaagJjgF9UPVRJY4AhFOkw58xMhWq0qdOAF44AchmKmVVguXNF7TqlRgYAYygBDAvCMhSAZxKnsALkQA6KgG5UV2yQBGTgTd5OAd3OAB6cLgASRIbAAsQaDA4X3hTUiwCAkIXVCA

**Related Issues:**

**Comment by HenryCharlesAnderson** [comment]
Just ran into this in my react project. A parent component defined a callback which expects a CONSTANT_CASE string enum. The child called the callback with a PascalCase string enum. 

At first I thought it was just an issue with the react component typing, somehow, but here's an equivalent example in plain ts: https://codesandbox.io/s/similar-enum-ts-issue-qrqv4

Not as minimal as the example in the OP, but I thought I'd share it anyway just in case it helps 🤷

**Comment by kyeotic** [comment]
I actually have the reverse issue, and I wish I had a way to make it not an error. I have two [identical enums](https://www.typescriptlang.org/play?#code/KYOwrgtgBAMglgIwCoE8AOwoG8BQUoBicATgM4AuUAvFAIwA0eUAysAMYD2IAJtVAEyN8SABYleNAMw4AvjhyhIUAIJo0qDNiZEylGgyatOPPoKajxfaXJwAbYJQBmHDgC5YiDZhrxk6YAB0OhQ4zhx8qur+QSQUQA) and an attempt to assign a value from one to the other fails.

In my actual case these two enums have identical names and identical values; they are from two different libraries. I can't use output from one to assign to the other, even though it is completely valid value-wise, because typescript sees them as different enums.

How can I get this to not be an error, as yours is?

**Comment by HenryCharlesAnderson** [comment]
> I actually have the reverse issue, and I wish I had a way to make it not an error. I have two [identical enums](https://www.typescriptlang.org/play?#code/KYOwrgtgBAMglgIwCoE8AOwoG8BQUoBicATgM4AuUAvFAIwA0eUAysAMYD2IAJtVAEyN8SABYleNAMw4AvjhyhIUAIJo0qDNiZEylGgyatOPPoKajxfaXJwAbYJQBmHDgC5YiDZhrxk6YAB0OhQ4zhx8qur+QSQUQA) and an attempt to assign a value from one to the other fails.
> 
> In my actual case these two enums have identical names and identical values; they are from two different libraries. I can't use output from one to assign to the other, even though it is completely valid value-wise, because typescript sees them as different enums.
> 
> How can I get this to not be an error, as yours is?

It's not my place to tell you how to write your code of course, but I wouldn't recommend relying on quirk of typescript since it can lead to bugs. As you can see in my previous comment, if one of your libraries changes the enum values (from numbers to strings, say), typescript won't necessary warn you. It might be better to write an adapter function that maps the library values to one that you create and control, and use your enum instead. That said, if this is really what you want, it should be possible.

If you give the enums the same name, it should work. Your exsmple fails because the two enums are named differently, I think.

[Here's an example of it working](https://codesandbox.io/s/charming-http-ndgtmc?file=/src/index.ts), using the NetworkStatus enum from apollo as an example of a library enum. 

What exactly were the libraries you were using? Are you sure the enums are **exactly** the same? It seems to throw an error if even one property name is different, as indicated by `inexactAppEnum` in the example.

**Comment by kyeotic** [comment]
> Are you sure the enums are exactly the same? It seems to throw an error if even one property name is different, as indicated by inexactAppEnum in the example.

As the [example I provided demonstrates](https://www.typescriptlang.org/play?#code/KYOwrgtgBAMglgIwCoE8AOwoG8BQUoBicATgM4AuUAvFAIwA0eUAysAMYD2IAJtVAEyN8SABYleNAMw4AvjhyhIUAIJo0qDNiZEylGgyatOPPoKajxfaXJwAbYJQBmHDgC5YiDZhrxk6YAB0OhQ4zhx8qur+QSQUQA) even completely identical enums produce an error. 

Unfortunately in my case I cannot use the library's enum because they did not export it. While this is almost certainly an oversight, I have no control over it. I can only adjust knobs that I can access, and this case the only one available seems to be a quick in typescript.

**Comment by jcalz** [comment]
Cross-linking to #29879

**Comment by Robert-Schirmer** [comment]
Should this be closed? The code example provided in the bug description is now fixed in v5.4.5 of typescript and errors as expected, [playground link](https://www.typescriptlang.org/play/?ts=5.4.5#code/LYewJgrgNgpgBAQTgbwFBw3GAPADiAJwBcsA7CYOADRWoEYAaagJjgF9UPVRJY4AhFOkw58xMhWq0qdOAF44AchmKmVVguXNF7TqlRgYAYygBDAvCMhSAZxKnsALkQA6KgG5UV2yQBGTgTd5OAd3OAB6cLgASRIbAAsQaDA4X3hTUiwCAkIXIA)
